pants-plugins/uses_services/st2cluster_rules.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2023 The StackStorm Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+from dataclasses import dataclass
+from textwrap import dedent
+
+from pants.backend.python.goals.pytest_runner import (
+    PytestPluginSetupRequest,
+    PytestPluginSetup,
+)
+from pants.backend.python.target_types import Executable
+from pants.backend.python.util_rules.pex import (
+    PexRequest,
+    VenvPex,
+    VenvPexProcess,
+    rules as pex_rules,
+)
+from pants.engine.fs import CreateDigest, Digest, FileContent
+from pants.engine.rules import collect_rules, Get, rule
+from pants.engine.process import FallibleProcessResult, ProcessCacheScope
+from pants.engine.target import Target
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+from uses_services.exceptions import ServiceMissingError
+from uses_services.platform_rules import Platform
+from uses_services.scripts.is_st2cluster_running import (
+    __file__ as is_st2cluster_running_full_path,
+)
+from uses_services.target_types import UsesServicesField
+
+
+@dataclass(frozen=True)
+class UsesSt2ClusterRequest:
+    """One or more targets need a running st2 cluster with all st2* services."""
+
+    auth_port: int = 9100
```

Comment:
If someone changes these server-side settings via the conf file, then I don't have a good way to detect--from the plugin--which file they used. Many tests make use of `ST2_CONFIG_PATH`, so that seems like a dangerous env var to allow into all tests using `pants.toml` `[tests].extra_env_vars`. I suppose we could allow `ST2_CONF`, since `launchdev.sh` and `st2ctl` also support using that.

Maybe I'll allow the `ST2_CONF` env var ONLY for the `uses=st2cluster` tests. But, I would have to parse the conf file here to figure out what host+port(s) to check for connectivity.

Then there are the client-side vars that someone would have to set to impact the st2cluster integration tests. Here's a chart of the client-side env vars and the analogous server-side vars. Should the tests use the server-side vars or the client-side vars? Or somehow use both, comparing to make sure they're compatible?

| client                            | server                                                                                           | notes                                                          |
|-----------------------------------|--------------------------------------------------------------------------------------------------|----------------------------------------------------------------|
| `ST2_CONFIG_FILE`                 | `ST2_CONFIG_PATH`                                                                                | Point to DIFFERENT conf files                                  |
| `ST2_BASE_URL`                    | `ST2_WEBUI__WEBUI_URL`                                                                           |                                                                |
| `ST2_API_URL`                     | `ST2_AUTH__API_URL`<br>`http{'s' if ST2_API_USE_SSL else ''}://{{ST2_API__HOST}:{ST2_API__PORT}` | server-side has two alternatives for setting this              |
| `ST2_AUTH_URL`                    | `http://{ST2_AUTH__HOST}:{ST2_AUTH__PORT}`                                                       |                                                                |
| `ST2_STREAM_URL`                  | `http://{ST2_STREAM__HOST}:{ST2_STREAM__PORT}`                                                   |                                                                |
| `ST2_AUTH_TOKEN`<br>`ST2_API_KEY` |                                                                                                  | client-side has 2 alternatives; server-side has none.          |
| `ST2_API_VERSION`                 |                                                                                                  | No reason to override this right now as we only support `v1`.  |
| `ST2_CACERT`                      |                                                                                                  | I'm not sure if someone would need to override this for tests. |